File: test7C#/test7/Validations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Console;

namespace test7
{
    public static class Validations
    {
        public static int WaitForValidInput(int nbOfChoices)
        {
            for (; ; )
            {
                if (int.TryParse(ReadLine(), out int choice))
                {
                    if (choice > 0 && choice <= nbOfChoices)
                    {
                        return choice;
                    }
                }

                WriteLine("Sorry, we didn't get that, can you select the number again?");
            }
        }

        public static int WaitForInt()
        {
            for (; ; )
            {
                if (int.TryParse(ReadLine(), out int integer))
                {
                    return integer;
                }

                WriteLine("Please answer with an integer.");
            }
        }
    }
}

File: test7Go/oop/Validations.go
package oop

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func WaitForValidInput(nbOfChoices int) int {
	reader := bufio.NewReader(os.Stdin)

	for {
		choiceStr, e := reader.ReadString('\n')
		if e == nil {
			choice, e2 := strconv.Atoi(string(choiceStr[0]))
			if e2 == nil {
				if choice > 0 && choice <= nbOfChoices {
					return choice
				}
			}
		}

		fmt.Print("Sorry, we didn't get that, can you select the number again?")
	}
}

func WaitForInt() int {
	reader := bufio.NewReader(os.Stdin)

	for {
		choiceStr, e := reader.ReadString('\n')
		if e == nil {
			choice, e2 := strconv.Atoi(string(choiceStr[0]))
			if e2 == nil {
				return choice
			}
		}

		fmt.Print("Please answer with an integer.")
	}
}
